src/main/kotlin/de/cyface/collector/handler/AuthorizationHandler.kt

```diff
@@ -63,159 +55,29 @@ class AuthorizationHandler(
 
-            // Before async operations, pause request body parsing to not lose the body or protocol upgrades.
-            strategy.pause(request)
-            val authentication: Future<io.vertx.ext.auth.User> = authProvider.authenticate(principal)
-            authentication.onSuccess { user: io.vertx.ext.auth.User ->
-                try {
-                    strategy.resume(request)
+            // Inform next handler which user is authenticated
+            val credentials = Oauth2Credentials(context.user().principal())
+            val username = credentials.username
+            // The "sub" is the subject claim which represents the unique id of the authenticated user.
+            val uuid = context.user().principal().getString("sub")
+            val user = User(UUID.fromString(uuid), username)
+            context.put("logged-in-user", user)
 
-                    // Before async operations, pause request body parsing to not lose the body or protocol upgrades.
-                    strategy.pause(request)
-                    // Load principal from authentication result, so it also contains the roles
-                    val loadUsers: Future<Set<User>> =
-                        loadAccessibleUsers(user.principal())
-                    loadUsers.onSuccess { accessibleUsers: Set<User> ->
-                        try {
-                            strategy.resume(request)
-                            LOGGER.trace(
-                                "Request for {} authorized to access users {}",
-                                username,
-                                accessibleUsers
-                            )
-                            val matched = accessibleUsers.stream().filter {
-                                    u: User ->
-                                u.name == username
-                            }.collect(Collectors.toList())
-                            Validate.isTrue(matched.size == 1)
-                            // Make sure it's the matched user
-                            context.put("logged-in-user", matched.stream().findFirst().get())
-                            context.put("accessible-users", accessibleUsers).next()
-                        } catch (e: RuntimeException) {
-                            context.fail(e)
-                        }
-                    }
-                    loadUsers.onFailure { e: Throwable ->
-                        LOGGER.error("Loading accessible users failed for user {}", username, e)
-                        context.fail(HTTPStatus.SERVER_ERROR, e)
-                    }
-                } catch (e: RuntimeException) {
-                    context.fail(HTTPStatus.SERVER_ERROR, e)
-                }
-            }
-            authentication.onFailure { e: Throwable ->
-                LOGGER.error("Authorization failed for user {}!", username)
-                context.fail(HTTPStatus.UNAUTHORIZED, e)
-            }
+            context.next()
```

Comment:
I did put this in a new line as it was hard to see at the end of the previous line:

`context.put("accessible-users", accessibleUsers).next()`